tests/two_queue_tests.cpp

```diff
@@ -0,0 +1,211 @@
+#define BOOST_TEST_MODULE appbase_two_queue_executor_test
+#include <boost/test/included/unit_test.hpp>
+#include <thread>
+#include <iostream>
+
+#include <appbase/two_queue_executor.hpp>
+
+using namespace appbase;
+
+BOOST_AUTO_TEST_SUITE(executor_test)
+
+std::thread start_app_thread(appbase::scoped_app& app) {
+   const char* argv[] = { boost::unit_test::framework::current_test_case().p_name->c_str() };
+   BOOST_CHECK(app->initialize(sizeof(argv) / sizeof(char*), const_cast<char**>(argv)));
+   app->startup();
+   std::thread app_thread( [&]() {
+      app->exec();
+   } );
+   return app_thread;
+}
+
+// verify functions only from default queue1 are executed when execution mode is not explictly set
+BOOST_AUTO_TEST_CASE( execute_from_default_queue ) {
+   appbase::scoped_app app;
+   auto app_thread = start_app_thread(app);
+   
+   // post functions
+   std::map<int, int> rslts {};
+   int seq_num = 0;
+   app->executor().post( priority::medium, app->executor().queue1(), [&]() { rslts[0]=seq_num; ++seq_num; } );
+   app->executor().post( priority::medium, app->executor().queue1(), [&]() { rslts[1]=seq_num; ++seq_num; } );
+   app->executor().post( priority::high,   app->executor().queue2(), [&]() { rslts[2]=seq_num; ++seq_num; } );
+   app->executor().post( priority::high,   app->executor().queue1(), [&]() { rslts[3]=seq_num; ++seq_num; } );
+   app->executor().post( priority::low,    app->executor().queue1(), [&]() { rslts[4]=seq_num; ++seq_num; } );
+   app->executor().post( priority::low,    app->executor().queue2(), [&]() { rslts[5]=seq_num; ++seq_num; } );
+   app->executor().post( priority::highest,app->executor().queue1(), [&]() { rslts[6]=seq_num; ++seq_num; } );
+   app->executor().post( priority::high,   app->executor().queue2(), [&]() { rslts[7]=seq_num; ++seq_num; } );
+
+   // stop application. Use lowest at the end to make sure this executes the last
+   app->executor().post( priority::lowest, app->executor().queue1(), [&]() {
+      // queue1 should have current function and queue2 should have all its functions
+      BOOST_REQUIRE_EQUAL( app->executor().queue1().size(), 1);
+      BOOST_REQUIRE_EQUAL( app->executor().queue2().size(), 3 );
```

Comment:
BOOST TEST is not thread safe: https://www.boost.org/doc/libs/1_81_0/libs/test/doc/html/boost_test/section_faq.html#boost_test.section_faq.is_boost_test_thread_safe

You should only use BOOST_REQUIRE_EQUAL on the main thread.

Reply:
Thanks. In this case when the two `BOOST_REQUIRE_EQUAL`s on the app thread are running,  the `BOOST_REQUIRE_EQUAL`s on the main thread have not started running yet. No conflicts would occur between the two threads.